src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -10,6 +10,69 @@ namespace System.IO
+        /// <exception cref="ArgumentNullException"><paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="bufferSize" /> is negative or zero.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="mode" />, <paramref name="rights" />, <paramref name="share" /> or <paramref name="options" /> contain an invalid value.
+        /// </exception>
+        /// <remarks><format type="text/markdown"><![CDATA[
+        /// ## Remarks
+        ///
+        /// This method can also throw any of the exceptions that <xref:System.IO.FileStream> can throw.
+        /// ]]></format></remarks>
+        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+        {
+            string badArg = null;
+
+            if (mode < FileMode.CreateNew || mode > FileMode.Append)
+                badArg = nameof(mode);
+
+            if (rights < FileSystemRights.ReadData || rights > FileSystemRights.FullControl)
+                badArg = nameof(rights);
+
+            if (share < FileShare.None || share > FileShare.Inheritable)
+                badArg = nameof(share);
+
+            if (options < FileOptions.None || options > FileOptions.Asynchronous)
+                badArg = nameof(options);
+
+            if (badArg != null)
+                throw new ArgumentOutOfRangeException(badArg, SR.ArgumentOutOfRange_Enum);
+
+            if (bufferSize <= 0)
+                throw new ArgumentOutOfRangeException(nameof(bufferSize));
+
+            if (fileSecurity == null)
+                throw new ArgumentNullException(nameof(fileSecurity));
+
+            fileInfo.SetAccessControl(fileSecurity);
+            return new FileStream(fileInfo.FullName, mode, mode == FileMode.Append ? FileAccess.Write : FileAccess.ReadWrite, share, bufferSize, options);
+        }
+
+        /// <summary>Creates a new directory, ensuring it is created with the specified directory security.</summary>
+        /// <param name="directoryInfo">The object describing a directory that does not exist in disk yet.</param>
+        /// <param name="directorySecurity">An object that determines the access control and audit security for the directory.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="directorySecurity" /> is <see langword="null" />.</exception>
+        public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity directorySecurity)
+        {
+            if (directorySecurity == null)
+                throw new ArgumentNullException(nameof(directorySecurity));
+
+            // FileSystem is an internal class in another assembly. How do I consume it here?
+            // FileSystem.CreateDirectory(directoryInfo.FullName, directorySecurity.GetSecurityDescriptorBinaryForm());
```

Comment:
I tried adding FileSystem.Windows.cs directly in FileSystem.AccessControl.csproj, but that file has a lot of additional dependencies. I wouldn't mind adding them if they were all in Common, but it also depends on other internal classes like PathInternal. So I'm not sure if I should continue pursuing that option, or if there's a better way.